juju/juju
# Simplify agent upgrade logic

Simplify agent upgrade logic:
- hosted models get the exact controller version unless otherwise specified
- controllers default to using the most recent minor version
- remove unused FindTools methods
- remove unused MinorVersion param

With hosted models you can just run "juju upgrade-model" and whatever the controller version is, even if a major version bump, that will be used.
With the controller model, automatically choose the most recent minor version.

The net effect is that using `--agent-version` arg to upgrade-model or upgrade-controller will not be needed any more except in edge cases.

We we also passing around unused MinorVersion params, so delete these. There's an api FindTools on the client facade still used by 2.9 clients - this can be removed in 3.1.

## Checklist


## QA steps

bootstrap a 2.9.37 controller and add a model
bootstrap a 3.0.3 controller
migrate the 2.9.37 model to the new 3.0.3 controller
juju model-upgrade and the hosted model goes from 2.9.37->3.0.3
juju upgrade-controller --build-agent and check that the controller is upgraded
